services/destination-debugger/eventDeliveryStatusUploader.go

```diff
@@ -0,0 +1,224 @@
+		res[job.DestinationID] = arr
+	}
+
+	rawJSON, err := json.Marshal(res)
+	if err != nil {
+		logger.Debugf(string(rawJSON))
+		misc.AssertErrorIfDev(err)
+		return
+	}
+
+	client := &http.Client{}
+	url := fmt.Sprintf("%s/dataplane/eventDeliveryStatus", configBackendURL)
+
+	retryCount := 0
+	var resp *http.Response
+	//Sending event schema to Config Backend
+	for {
+
+		req, err := http.NewRequest("POST", url, bytes.NewBuffer([]byte(rawJSON)))
+		if err != nil {
+			misc.AssertErrorIfDev(err)
+			return
+		}
+		req.Header.Set("Content-Type", "application/json;charset=UTF-8")
+		req.SetBasicAuth(config.GetWorkspaceToken(), "")
+
+		resp, err = client.Do(req)
+		if err != nil {
+			logger.Error("Config Backend connection error", err)
+			if retryCount > maxRetry {
+				logger.Errorf("Max retries exceeded trying to connect to config backend")
+				return
+			}
+			retryCount++
+			time.Sleep(retrySleep)
+			//Refresh the connection
+			continue
+		}
+		break
+	}
+
+	if !(resp.StatusCode == http.StatusOK ||
+		resp.StatusCode == http.StatusBadRequest) {
+		logger.Errorf("Response Error from Config Backend: Status: %v, Body: %v ", resp.StatusCode, resp.Body)
+	}
+}
+
+func handleJobs() {
+	deliveryStatusesBuffer = make([]*DeliveryStatusT, 0)
+
+	for {
+		select {
+		case deliveryStatus := <-deliveryStatusesBatchChannel:
+			deliveryStatusesBufferLock.Lock()
+
+			//If deliveryStatusesBuffer size is more than maxESQueueSize, Delete oldest.
+			if len(deliveryStatusesBuffer) > maxESQueueSize {
+				deliveryStatusesBuffer[0] = nil
+				deliveryStatusesBuffer = deliveryStatusesBuffer[1:]
+			}
+
+			//Append to request buffer
+			deliveryStatusesBuffer = append(deliveryStatusesBuffer, deliveryStatus)
+
+			deliveryStatusesBufferLock.Unlock()
+		}
+	}
+}
+
+func flushJobs() {
+	for {
+		select {
+		case <-time.After(batchTimeout):
+			deliveryStatusesBufferLock.Lock()
+
+			flushSize := len(deliveryStatusesBuffer)
+			var flushJobs []*DeliveryStatusT
+
+			if flushSize > maxBatchSize {
+				flushSize = maxBatchSize
+			}
+
+			if flushSize > 0 {
+				flushJobs = deliveryStatusesBuffer[:flushSize]
+				deliveryStatusesBuffer = deliveryStatusesBuffer[flushSize:]
+			}
+
+			deliveryStatusesBufferLock.Unlock()
+
+			if flushSize > 0 {
+				uploadJobs(flushJobs)
+			}
+
+			flushJobs = nil
+		}
+	}
+}
+
+func updateConfig(sources backendconfig.SourcesT) {
+	configSubscriberLock.Lock()
+	uploadEnabledDestinationIDs = []string{}
+	for _, source := range sources.Sources {
+		for _, destination := range source.Destinations {
+			if destination.Config != nil {
+				if destination.Enabled && destination.Config.(map[string]interface{})["eventDelivery"] == true {
+					uploadEnabledDestinationIDs = append(uploadEnabledDestinationIDs, destination.ID)
+				}
+			}
+		}
+	}
+	configSubscriberLock.Unlock()
+}
+
+func backendConfigSubscriber() {
+	configChannel := make(chan utils.DataEvent)
+	backendconfig.Subscribe(configChannel, "processConfig")
```

Comment:
Should this be backendConfig instead of processConfig, since router and batchrouter process events even when isProcessorEnabled is false?

Reply:
done